### PR TITLE
Feature/new qos [5481]

### PR DESCRIPTION
**Pull request by LuisGP**
Added Disable Positive ACKs, Deadline and Lifespan QoS.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -48,14 +48,19 @@ The Publish Window allows you to generate a new Shape and corresponding RTPS Top
...
+* Ownership: The Ownership Qos determines whether the instance (color) of the Topic (Shape) is owned by a single Publisher. If the selected ownership is EXCLUSIVE the Publisher will use the Ownership strength value as the strength of its publication. Only the publisher with the highest strength can publish in the same Topic with the same Key.
+* Deadline: The Deadline Qos determines the maximum expected amount of time between samples. When the deadline is missed the application will be notified and a message printed on the console.
+* Lifespan: The Lifespan Qos determines the duration while the sample is still valid. When a sample's lifespan expires, it will be removed from publisher and subscriber histories.
+* Disable Positive ACKs: If enabled, writers don't expect positive ACKNACK messages from the subscriber, considering delivered sent samples after the indicated duration if no negative ACKNACK was received.
```
Comment (raquelalvarezbanos) [inline README.md]: After talking to Jaime he thinks it is better to not add Disable Positive ACKs to shapes demo, as it is not a standard DDS QoS. Can we please remove it? And apologies for the work that had already been done

File: README.md [hunk 2]
```diff
@@ -65,8 +70,10 @@ The Subscribe Window allows you to read and represent existing shapes on the net
 
 This menu provides the options present in the Publish Windows plus the following additions:
 
-* Time Based Filter: This value can be used to specify the minimum amount of time (in milliseconds) that the subscriber wants between updates. 
-* Content Based Filter: This filter draws a rectangle in the shapes window. Only the instances that are included in this rectangle are accepted. The rest of them are ignored. The user can dynamically resize and move this content filter. 
+* Time Based Filter: This value can be used to specify the minimum amount of time (in milliseconds) that the subscriber wants between updates.
+* Content Based Filter: This filter draws a rectangle in the shapes window. Only the instances that are included in this rectangle are accepted. The rest of them are ignored. The user can dynamically resize and move this content filter.
+
+Take into acoount that Disabling positive ACKs and modify Lifespan will not have any visual effect.
```
Comment (raquelalvarezbanos) [inline README.md]: I think `Deadeline` and `Lifespan` should also be described in this section, as these QoS are also applicable on the subscriber side. You can repeat the same as in the above section or mention that they work in a similar way

File: forms/ui/moc_DrawArea.cpp [hunk 3]
```diff
@@ -1,90 +0,0 @@
-/****************************************************************************
```
Comment (raquelalvarezbanos) [inline forms/ui/moc_DrawArea.cpp]: This and the files above were deleted as now we use the moc compiler, is that correct?

File: src/shapesdemo/ShapePublisher.cpp [hunk 4]
```diff
@@ -74,3 +76,11 @@ void ShapePublisher::onPublicationMatched(Publisher* /*pub*/, rtps::MatchingInfo
     else if(info.status == rtps::REMOVED_MATCHING)
         std::cout << "Publisher  in topic " << m_attributes.topic.getTopicName() << " REMOVES Sub: " << info.remoteEndpointGuid << "*****************************" << std::endl;
 }
+
+void ShapePublisher::on_offered_deadline_missed(
+        Publisher*,
+        const OfferedDeadlineMissedStatus&)
+{
+    //std::cout << "Offered deadline missed" << std::endl;
```
Comment (raquelalvarezbanos) [inline src/shapesdemo/ShapePublisher.cpp]: Can we remove this comment and the one in ShapeSubscriber.cpp, please?

File: forms/ui/moc_DrawArea.cpp (hunk 3 above)
Comment (LuisGP) [reply forms/ui/moc_DrawArea.cpp]: Exactly.

File: src/shapesdemo/ShapeSubscriber.cpp [hunk 5]
```diff
@@ -63,7 +63,7 @@ void ShapeSubscriber::onNewDataMessage(Subscriber* sub)
     Shape shape;
     shape.m_type = this->m_shapeType;
     SampleInfo_t info;
-    while(sub->takeNextData((void*)&shape.m_shape,&info))
+    while(sub->readNextData((void*)&shape.m_shape,&info))
```
Comment (raquelalvarezbanos) [inline src/shapesdemo/ShapeSubscriber.cpp]: Why was this changed? Maybe you had this change locally and committed it by accident or is there a reason to change it?
